lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -0,0 +1,411 @@
+        else:
+            cell[...] = np.exp(-6147.795166 / cell + 24.31720686)
+    result = temp.copy(data=data)
+    result.units = Unit('hPa')
+    result.convert_units('Pa')
+    result.rename("saturated_vapour_pressure")
+    return result
+
+
+def humidity_ratio_fm_rh(temperature, relative_humidity, pressure):
+    ''' Function to compute humidity ratio given temperature pressure and
+    relative humidity.
+
+    Args:
+        temperature : iris.cube.Cube
+            Cube of temperature which will be converted to celsius
+            prior to calculation.
+        rel_humidity : iris.cube.Cube
+            Cube of relative humidity.
+        pressure : iris.cube.Cube
+            Cube of pressure which will be converted to kilopPascals
+            prior to calculation.
+
+    Returns
+        hr : iris.cube.Cube
+            humidity ratio cube with units (1).
+
+    References:
+        ASHRAE Fundamentals handbook (2005) Equation 22, 24, p6.8
+    '''
+    # Decouple local variables from variables supplied
+    # Ensure that variables are in correct units
+    temp = temperature.copy()
+    temp.convert_units('celsius')
+    rh = relative_humidity.copy()
+    rh.convert_units(1)
+    press = pressure.copy()
+    press.convert_units('kPa')
+    svp = saturation_vapour_pressure_goff_gratch(temp, press)
+    svp.convert_units('kPa')
+
+    # Calculation
+    result_numer = (0.62198 * rh.data * svp.data)
+    result_denom = (press.data - (rh.data * svp.data))
+    hr = temp.copy(data=result_numer / result_denom)
+
+    # Tidying up cube
+    hr.rename("Humidity ratio")
+    hr.units = Unit("1")
+    return hr
+
+
+def humidity_ratio_fm_wb(temperature, wet_bulb, pressure):
+    ''' Function to compute humidity ratio given a temperature pressure and
+    wet bulb temperature.
+
+    Args:
+        temp: iris.cube.Cube
+            Cube of temperature which will be converted to Celsius
+            prior to calculation
+
+        wet_bulb: iris.cube.Cube
+            Cube of wet_bulb_temperature which will be converted to Celsius
+            prior to calculation
+
+        pressure: iris.cube.Cube
+            Cube of pressure which will be converted to kiloPascals
+            prior to calculation
+
+    Returns:
+        humidity_ratio : iris.cube.Cube
+            humidity ratio cube with units(1)
+
+    References:
+        ASHRAE Fundamentals handbook (2005) Equation 22, 24, p6.8
+    '''
+    temp = temperature.copy()
+    temp.convert_units("celsius")
+    press = pressure.copy()
+    press.convert_units("kPa")
+    wb = wet_bulb.copy()
+    wb.convert_units("celsius")
+
+    # Calculation
+    wb_depression = temp - wb
+    svp = saturation_vapour_pressure_goff_gratch(temp, press)
+    svp.convert_units('kPa')
+    ws = 0.62198 * svp / (press - svp)
+    # This is inefficient because it produces both answers
+    above_zero = (((2501. - 2.326 * wb.data) * ws.data -
```

Comment:
You need to indicate that these are equations 35/37 in the ASHRAE document, both here and in the docstring.